src/main/java/me/jellysquid/mods/sodium/client/render/chunk/lists/SortedRenderLists.java

```diff
@@ -44,7 +48,17 @@ public void add(RenderSection section) {
             list.add(section);
         }
 
-        public SortedRenderLists build() {
+        public SortedRenderLists build(Vec3d cameraPos) {
+            //noinspection ForLoopReplaceableByForEach
+            for (int i = 0; i < this.lists.size(); i++) {
+                ChunkRenderList renderList = this.lists.get(i);
+                RenderRegion region = renderList.getRegion();
+                double dx = cameraPos.x - region.getCenterX();
+                double dy = cameraPos.y - region.getCenterY();
+                double dz = cameraPos.z - region.getCenterZ();
+                renderList.setDistanceFromCamera((dx * dx) + (dy * dy) + (dz * dz));
+            }
+            this.lists.sort(LIST_DISTANCE_COMPARATOR);
```

Comment:
it might be faster to use `unstableSort` here because that can use quicksort which is allocation free. I can't think of a reason for a stable sort to be necessary. `ObjectArrayList` does a stable sort by default.

Reply:
Is this implementation of quicksort O(n) on sorted inputs? I think `sort` uses timsort which should have that behavior.

Reply:
yeah the default `Arrays.sort` that it uses for stable sort might just be better. The quicksort  is kinda hard to read so I'm not sure it'd actually fare better.